server/src/main/java/org/opensearch/common/joda/JodaDeprecationPatterns.java

```diff
@@ -69,11 +67,9 @@ public static boolean isDeprecatedPattern(String pattern) {
         List<String> patterns = DateFormatter.splitCombinedPatterns(pattern);
 
         for (String subPattern : patterns) {
-            boolean isDeprecated = subPattern.startsWith("8") == false && FormatNames.exist(subPattern) == false &&
-                JODA_PATTERNS_DEPRECATIONS.keySet().stream()
-                                          .filter(s -> subPattern.contains(s))
-                                          .findAny()
-                                          .isPresent();
+            boolean isDeprecated = subPattern.startsWith("8") == false
+                && FormatNames.exist(subPattern) == false
+                && JODA_PATTERNS_DEPRECATIONS.keySet().stream().filter(s -> subPattern.contains(s)).findAny().isPresent();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It is more convenient and readable to directly use Stream::anyMatch instead of Stream::filter, Stream::findAny, and Optional::isPresent when the matched data returned by Stream::findAny is not used anywhere else.